wpimath/src/main/native/cpp/controller/PIDController.cpp

```diff
@@ -56,6 +56,20 @@ PIDController::PIDController(double Kp, double Ki, double Kd,
   wpi::SendableRegistry::Add(this, "PIDController", instances);
 }
 
+PIDController::PIDController(PIDConstants constants) {
+  m_Kp = constants.Kp;
+  m_Ki = constants.Ki;
+  m_Kd = constants.Kd;
+  m_period = constants.period;
+  m_iZone = constants.IZone;
+  static int instances = 0;
```

Comment:
This will use a different counter from the normal constructor. The cleanest solution would be to defer to that constructor. (Especially since `PIDConstants` can be modified after construction, so we need to check the values anyways)